packages/lib/src/storybook/StoryMeta.types.ts

```diff
@@ -0,0 +1,93 @@
+import { Meta, StoryFn, StoryObj } from '@storybook/react';
+import { ComponentProps } from 'react';
+import DarkModeProps from '../DarkModeProps';
+
+import { StoryArgType } from './storybookArgTypes';
+
+// Re-defining LG provider prop keys here since importing from the package
+// will cause circular dependencies
+export interface LeafyGreenProviderProps extends DarkModeProps {
+  baseFontSize?: number;
+}
+
+export interface ControlsConfig {
+  /**
+   * Props excluded from controls
+   */
+  exclude: Array<string>;
+}
+
+/**
+ * Parameters for Chromatic
+ *
+ * Compiled from https://mongodb.chromatic.com/docs/
+ */
+export interface ChromaticConfig {
+  /** https://mongodb.chromatic.com/docs/ignoring-elements#ignore-stories */
+  disableSnapshot?: boolean;
+  /** https://mongodb.chromatic.com/docs/delay#delay-a-story */
+  delay?: number;
+  /** https://mongodb.chromatic.com/docs/animations#css-animations */
+  pauseAnimationAtEnd?: boolean;
+  /** https://mongodb.chromatic.com/docs/threshold#setting-the-threshold */
+  diffThreshold?: number;
+  /** https://mongodb.chromatic.com/docs/threshold#anti-aliasing */
+  diffIncludeAntiAliasing?: boolean;
+  [key: string]: any;
+}
+
+/**
+ * Story Parameters
+ */
+type StoryParameters<T extends React.ElementType> = Meta<T>['parameters'] & {
+  /**
+   * The default story to be displayed on `mongodb.design`.
+   * Explicitly exclude a default story by setting this to `null`
+   */
+  default: string | null;
+
+  controls?: ControlsConfig;
+
+  chromatic?: ChromaticConfig;
+};
+
+/**
+ * Story control arg types
+ */
+type ArgTypes<
+  T extends React.ElementType,
+  XP extends Record<string, any> = {},
+> = Partial<
+  | {
+      [key in keyof ComponentProps<T>]: StoryArgType;
+    }
+  | {
+      [key in keyof LeafyGreenProviderProps]: StoryArgType;
+    }
+  | {
+      [key in keyof XP]: StoryArgType;
+    }
+>;
+
+/**
+ * Type of the Story default export
+ */
+export type StoryMetaType<
+  T extends React.ElementType,
+  XP extends Record<string, any> = {},
+> = Omit<Meta<T>, 'component' | 'argTypes' | 'args'> & {
+  title?: string;
+  component?: T;
+  parameters: StoryParameters<T>;
+  argTypes?: ArgTypes<T, XP>;
+  args?: Partial<ComponentProps<T> | LeafyGreenProviderProps | XP>;
```

Comment:
As I understand, these props should also be defined in `Meta`. Why are they duplicated here?

At the very least, it feels like these should be commented with why the duplication was necessary.

Reply:
It's typed as `Record<string, any>`